android/app/src/main/java/org/fossasia/openevent/core/main/MainActivity.java

```diff
@@ -831,47 +738,52 @@ public void downloadFromAssets() {
         fromServer = false;
         if (!SharedPreferencesUtil.getBoolean(ConstantStrings.DATABASE_RECORDS_EXIST, false)) {
             //TODO: Add and Take counter value from to config.json
-            SharedPreferencesUtil.putBoolean(ConstantStrings.DATABASE_RECORDS_EXIST, true);
 
             startDownloadListener();
             Timber.d("JSON parsing started");
 
             StrategyRegistry.getInstance().getEventBusStrategy().postEventOnUIThread(new CounterEvent(7)); // Bump if increased
 
-            readJsonAsset(Urls.EVENT);
-            readJsonAsset(Urls.SESSIONS);
-            readJsonAsset(Urls.SPEAKERS);
-            readJsonAsset(Urls.TRACKS);
-            readJsonAsset(Urls.SPONSORS);
-            readJsonAsset(Urls.MICROLOCATIONS);
-            readJsonAsset(Urls.SESSION_TYPES);
-            //readJsonAsset(Urls.FAQS);
+            disposable.add(downloadHandler.downloadFromAsset()
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe((name) -> {
+                        switch (name) {
```

Comment:
It will be refactor to following when ww will remove event dependency:
`disposable.add(downloadHandler.downloadFromAsset()
                    .subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribe(completeDownloadHandler::onDownloadDone);`